app/models/review.rb

```diff
@@ -0,0 +1,9 @@
+class Review < ApplicationRecord
+  MAX_STAR_RATING = 3
+  NO_STAR_RATING = 0
+
+  belongs_to :reviewable, polymorphic: true
```

Comment:
🎉

Reply:
That's an interesting design choice. Can you please elaborate?

Reply:
Since reviews tend to come on multiple objects, I wanted to make it "future ready".
A restaurant has reviews but then other futuristic objects (e.g., specific dishes for each restaurant can have reviews) can also have reviews and the fields are pretty generic (rateing, comment & commenter).
It decouples a review from specific object.
The 'bad parts' arose after making the change for adding a rating column as suggested by @sagish  - it added some complexity and code I dislike... (can refer to commit 2e67f59 below), would love to hear your thoughts on the implementation of this change

Reply:
The issue I take with this design is:
1. it doesn't really maintain the kiss principle for now (in favor of future-proofing, so it's not too major)
2. We're creating a bidirectional coupling (review <-> reviewable) here which is less advisable.